refs #371: Added the whole contributions guidelines section (#372)
carenalgas/popochiu: docs/src/contributing-to-popochiu/contributing-documentation/basic-rules.md

!!! tip "Updating visual assets"
    When the editor interface changes or get some new feature, screenshots or other visuals may need to be updated.  
    Taking consistent screenshots is a tough job: we recommend using the **Popochiu Sample Game**. Clone the repo and update the `addons/popochiu` folder with your changes.

    Remember to annotate the screenshots if it adds to clarity and readability of your explanation. If possible use Godot dark theme and red annotations for consistency.

---

## Understanding Documentation Types

Before you start writing, it's important to understand what type of documentation you’re contributing to. Popochiu’s documentation broadly follows the categories defined by the [Diátaxis framework](https://diataxis.fr/), by which content fall into one of four types (each with a specific _user's main goal_ in brackets):

- **Tutorials**: Step-by-step guides that show how stuff work (**learning**)

> - **How-to Guides**: Focused, task-oriented instructions to achieve a certain goal . think recipes (**achieving a goal**).  

Reviewer: Not sure about my suggestion, but the point is that I think this sentence needs something to make it clearer. Again, I asked ChatGPT for help.

```suggestion
- **How-to Guides**: Focused, task-oriented instructions designed to achieve a specific goal—think of them as recipes. (Main goal: **achieving a goal**)
```

> - **Explanations**: Conceptual overviews and background information (main goal: **understanding**)
> - **Reference**: Well-organized list of techical details and descriptions (main goal: **finding information**)

Reviewer: Based on what ChatGPT suggested for the **How-to Guides** point.

```suggestion
- **Explanations**: Conceptual overviews and background information. (Main goal: **understanding**)
- **Reference**: Well-organized list of techical details and descriptions. (Main goal: **finding information**)
```

Each section in Popochiu's documentation aligns with a type from the Diátaxis framework. Here's a quick overview:

| **Section**              | **Content types**        | **Purpose**                                                                                                               |
|--------------------------|--------------------------|---------------------------------------------------------------------------------------------------------------------------|
| Getting Started          | Tutorials, Explanations  | Provides the basics for installing Popochiu, taking first steps, and joining the community.                               |
| The Editor Handbook      | References, Tutorials    | Showcases the functionality of the Popochiu plugin, explains how it works, and serves as a go-to guide for its use.       |
| The Engine Handbook      | References, Explanations | Explains the plugin's architecture and provides a comprehensive API reference for game developers.                        |
| How to Develop a Game    | Tutorials, How-To        | Guides users through setting up their first game, learning by doing, and building confidence to explore advanced features. |

> | Advanced Techniques      | Tutorials, How-To        | Offers in-depth guides and practical examples for getting the most out of Popochiu (and beyond).                          |

Reviewer: Based on this, one would think that the **Advanced Techniques** section should appear in the main navigation panel, but right now it’s a subcategory of the **How to Develop a Game** section. I’m not sure if this needs to be clarified or if the navigation panel needs to be updated.

Author: Ah! I'm an idiot! You're right, I totally forgot to reorganize the menu.

Thanks!

| Contributing to Popochiu | Explanations, References | Provides all the information contributors need to get involved in Popochiu development.                                    |

> If you add new stuff, make sure you put in in the right place. If in doubt, reach out for **@stickgrinder**.

Reviewer: ```suggestion
If you add new stuff, make sure you put it in the right place. If in doubt, reach out for **@stickgrinder**.
```

---

## Contributing to missing documentation

Some areas of the documentation are marked as **TODO**. If you think you know how to complete them (yay!):

1. Draft an outline for the missing content.
2. Commit the outline and open a draft PR.
3. Assign **@stickgrinder** as a reviewer.
4. Incorporate feedback and proceed with the full write-up.

> If you can't - or don't want to incorporate all the feedback (for example if we ask you to expand the docs too much and you are unable to provide all the necessary work), we can still cooperate to complete it together.

Reviewer: I love this!!! ❤️❤️❤️

Author: Thanks! I mentioned myself so I can own this and act as a proxy for you and you don't get too distracted from other stuff. I'll reach out for you anytime for validation, if that's something non-trivial, of couse.
